.github/workflows/release.yml

```diff
@@ -0,0 +1,129 @@
+    if: needs.checks.outputs.HAS_TAG == 'false'
+    steps:      
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Set Git author
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+      
+      - name: Get node version
+        id: get_node_version
+        run: |
+          echo "NVMRC=$(cat .nvmrc)" >> $GITHUB_OUTPUT
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "${{ steps.get_node_version.outputs.NVMRC }}"
+
+      - name: Install dependencies
+        run: yarn install
+        
+      - name: Build
+        run: yarn build
+
+      - name: Bump CLI version
+        run: npm version patch --force
+
+      - name: Read version from package.json
+        id: get_cli_version
+        run: echo "CLI_VERSION=$(node -p "require('./package.json').version")" >> $GITHUB_OUTPUT
+
+      - name: Get latest tag
+        id: get_latest_tag
+        run: echo "LATEST_TAG=$(git describe --abbrev=0 --tags)" >> $GITHUB_OUTPUT
+          
+      - name: Commit and push tag to main branch
+        run: |
+          git add .
+          git commit --amend -m "Release v${{ steps.get_cli_version.outputs.CLI_VERSION }}"
+          git tag -f ${{ steps.get_latest_tag.outputs.LATEST_TAG }}
+          git push origin main --follow-tags
+      
+      - name: Create GitHub Release
+        uses: ncipollo/release-action@v1
+        with:
+          name: "v${{ steps.get_cli_version.outputs.CLI_VERSION }}"
+          tag: "v${{ steps.get_cli_version.outputs.CLI_VERSION }}"
+          generateReleaseNotes: "true"
+          makeLatest: "true"
+          
+      - name: Publish to NPM
+        run: npm publish --access public  
+
+  update_doc:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write 
+    needs: release
+    env:
+      BRANCH_NAME: update-cli-version-to-v${{ needs.release.outputs.CLI_VERSION }}
+    steps:      
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          repository: DevCycleHQ/devcycle-docs
+          # need access token
+          token: ${{ secrets.PAT_TOKEN }}
+
+      - name: Set Git author
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+          
+      - name: Update CLI version in docs repo
+        run: |
+          ls -al
+          echo $BRANCH_NAME
+          git checkout -b "$BRANCH_NAME"
+          git branch
+          sed -i 's/const DVC_CLI_VERSION = .*/const DVC_CLI_VERSION = '\''v${{ needs.release.outputs.CLI_VERSION }}'\'' \/\/ auto updated by dvc cli release workflow/' docusaurus.config.js
+          git add docusaurus.config.js
+          git commit -m "Update CLI version to v${{ needs.release.outputs.CLI_VERSION }}"
```

Comment:
This should be a shell script that's parameterized to make it more manageable - and the outputs should be cleaned up to make sure that it's not messy in the output on GHA.